tests/api/test_result.py

```diff
@@ -16,30 +16,99 @@
 """
 import numpy as np
 import pytest
+from strawberryfields.backends.states import BaseGaussianState
 
 from strawberryfields.result import Result
 
 pytestmark = pytest.mark.api
 
 
+raw_results = {
+    "output": [np.ones((2, 3, 4, 5))],
+    0: [1, 2, 3],
+    1: [4, 5],
+    "other_data": [1, 0, 1, 2],
+    "string": "metadata",
+}
+
+
 class TestResult:
     """Tests for the ``Result`` class."""
 
+    def test_samples(self):
+        """Test that ``samples`` is correctly returned."""
+        result = Result(raw_results)
+        assert result.samples is not None
+        assert np.equal(result.samples, np.ones((2, 3, 4, 5))).all()
+
+    def test_raw(self):
+        """Test that raw results are correctly returned."""
+        result = Result(raw_results)
+        assert result.raw == raw_results
+
+    def test_samples_dict(self):
+        """Test that ``samples_dict`` is correctly returned."""
+        result = Result(raw_results)
+        assert result.samples_dict == {0: [1, 2, 3], 1: [4, 5]}
+
+    def test_ancilla_samples(self):
+        """Test that ancilla samples are correctly returned."""
+        ancilla_samples = {0: [0, 1], 2: [1, 3]}
+        result = Result(raw_results, ancilla_samples=ancilla_samples)
+        assert result.ancilla_samples == ancilla_samples
+
+    def test_state(self):
+        """Test that the state can be correctly set and returned."""
```

Comment:
This also tests that a state cannot be set twice (unless, of course, the state is set to `None`).